### ISSUE TITLE
SystemUI crashing

### ISSUE THREAD

**Issue by VytskaLT** [topjohnwu/Magisk#5116]
Device: OnePlus 8
Android version: 11
Magisk version name: 0c1f68816e9eb02fe35e4c36221c77d4d87ab33e
Magisk version code: 23016

For a while, Magisk worked perfectly, but then suddenly booting up takes much longer, when I unlock my phone it says "System UI isn't responding", can't turn on wifi or mobile data (says no sim card detected) and in a few minutes the phone crashes and turns on "Qualcomm Crashdump Mode".

Before this started happening, I had Riru and LSPosed modules.

I tried:

* Reinstalling Magisk with the flash zip and the patch image techniques
* Reinstalling LineageOS (without factory resetting)

The issues are completely gone when I uninstall Magisk.

magisk.log says:
```
12-25 00:19:16.021  2638  2638 I : Magisk 5b858b95(23016) daemon started
12-25 00:19:16.021  2638  2638 I : * Device API level: 30
12-25 00:19:16.033  2638  2653 I : ** boot_complete triggered
```

**Comment by osm0sis** [comment]
Delete /data/adb contents, likely you've got some rogue startup script. Not really a Magisk issue in that case though.

**Comment by VytskaLT** [comment]
Issue still persists. /data/adb is empty.

**Comment by osm0sis** [comment]
Your log shows 23000 but you reported 23016.

**Comment by VytskaLT** [comment]
> Your log shows 23000 but you reported 23016. Try 23016 with empty /data/adb

Oh, sorry about that.
This is the magisk.log with 23016:
```
04-25 23:41:55.577   709   709 I : Magisk 5b858b95(23016) daemon started
04-25 23:41:55.577   709   709 I : * Device API level: 30
04-25 23:41:55.589   709   711 I : ** post-fs-data mode running
04-25 23:41:55.589   709   711 I : * Mounting mirrors
04-25 23:41:55.589   709   711 I : link: /dev/LOaRK/.magisk/mirror/metadata
04-25 23:41:55.589   709   711 I : mount: /dev/LOaRK/.magisk/mirror/system_ext
04-25 23:41:55.589   709   711 I : mount: /dev/LOaRK/.magisk/mirror/product
04-25 23:41:55.589   709   711 E : mount /dev/LOaRK/.magisk/block/vendor->/dev/LOaRK/.magisk/mirror/vendor failed with 6: No such device or address
04-25 23:41:55.589   709   711 I : mount: /dev/LOaRK/.magisk/mirror/vendor
04-25 23:41:55.589   709   711 I : link: /dev/LOaRK/.magisk/mirror/persist
04-25 23:41:55.589   709   711 E : mount /dev/LOaRK/.magisk/block/vendor->/dev/LOaRK/.magisk/mirror/vendor failed with 22: Invalid argument
04-25 23:41:55.589   709   711 I : mount: /dev/LOaRK/.magisk/mirror/vendor
04-25 23:41:55.589   709   711 I : mount: /dev/LOaRK/.magisk/mirror/data
04-25 23:41:55.589   709   711 I : link: /dev/LOaRK/.magisk/mirror/system
04-25 23:41:55.589   709   711 I : mount: /dev/LOaRK/.magisk/mirror/system_root
04-25 23:41:55.589   709   711 I : * Initializing Magisk environment
04-25 23:41:55.602   709   711 I : * Running post-fs-data.d scripts
04-25 23:41:55.603   709   711 I : * Running module post-fs-data scripts
04-25 23:41:55.603   709   711 I : * Loading modules
12-25 16:56:40.319   709   711 I : ** late_start service mode running
12-25 16:56:40.320   709   711 I : * Running service.d scripts
12-25 16:56:40.320   709   711 I : * Running module service scripts
12-25 16:58:19.185   709   711 I : ** boot_complete triggered
```

**Comment by osm0sis** [comment]
You'll need to upload a last_kmsg or console-ramoops from after one of the reboots.

**Comment by VytskaLT** [comment]
> You'll need to upload a last_kmsg or console-ramoops from after one of the reboots.

I can't find any instructions on how to do that which worked for me
I tried `cat /sys/fs/pstore/console-ramoops` but the /sys/fs/pstore folder is empty.
`/proc/last_kmsg` doesn't exist either.
Apologies, I'm not very familiar with this stuff.

**Comment by canyie** [comment]
adb logcat

**Comment by VytskaLT** [comment]
> adb logcat

This is the logcat starting from device boot and ending at the crash: https://ufile.io/x4md7alk

**Comment by yujincheng08** [comment]
send `/data/anr/` and `/data/tomestone`. Zip and upload from github rather than a third party netdisk

**Comment by VytskaLT** [comment]
> send `/data/anr/` and `/data/tomestone`. Zip and upload from github rather than a third party netdisk
[anr.zip](https://github.com/topjohnwu/Magisk/files/7780265/anr.zip)
[tombstones.zip](https://github.com/topjohnwu/Magisk/files/7780269/tombstones.zip)
`/data/tomestone` didn't exist, so I assumed you meant /data/tombstones

**Comment by Unimog411** [comment]
On my  HTC One (M7) running Android 5.02, Magisk v24.1 crashed immediately after opening the app. In `adb logcat` I noted an error `JNI DETECTED ERROR IN APPLICATION`, which seems to be related to UI(-animations?).
Ununstalling v24.1 and installing v.23.0 solved my issue.

**Comment by canyie** [comment]
@Unimog411 Will installing the latest canary (24102) solved your issue? If not, open an issue and attach your log.

**Comment by Unimog411** [comment]
Thanks @canyie !
I'd love to try this if you tell me where to find "latest canary (24102)"  :-)

**Comment by Unimog411** [comment]
@canyie 
I found and tried "latest canary (24102)". Problem with "crashed immediately after opening the app" persists. From logcat:
```
I/DEBUG   (  318): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   (  318): Build fingerprint: 'htc/vodafone_uk/m7:5.0.2/LRX22G/476210.51:user/release-keys'
I/DEBUG   (  318): Revision: '3'
I/DEBUG   (  318): ABI: 'arm'
I/DEBUG   (  318): pid: 18743, tid: 18743, name: opjohnwu.magisk  >>> com.topjohnwu.magisk <<<
I/DEBUG   (  318): signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
I/DEBUG   (  318): Abort message: 'art/runtime/check_jni.cc:65] JNI DETECTED ERROR IN APPLICATION: can't call void android.view.View.setTranslationY(float) on null object'
I/DEBUG   (  318):     r0 00000000  r1 00004937  r2 00000006  r3 00000000
I/DEBUG   (  318):     r4 b6fd72a4  r5 00000006  r6 00000002  r7 0000010c
I/DEBUG   (  318):     r8 00000000  r9 b8b7d8c0  sl b8b87958  fp 00000caa
I/DEBUG   (  318):     ip 00004937  sp be86fa10  lr b6f6b7bd  pc b6f92240  cpsr 60070010
I/DEBUG   (  318): 
I/DEBUG   (  318): backtrace:
I/DEBUG   (  318):     #00 pc 0003a240  /system/lib/libc.so (tgkill+12)
I/DEBUG   (  318):     #01 pc 000137b9  /system/lib/libc.so (pthread_kill+52)
I/DEBUG   (  318):     #02 pc 000143d7  /system/lib/libc.so (raise+10)
I/DEBUG   (  318):     #03 pc 00010cdd  /system/lib/libc.so (__libc_android_abort+36)
I/DEBUG   (  318):     #04 pc 0000f460  /system/lib/libc.so (abort+4)
I/DEBUG   (  318):     #05 pc 0021aa11  /system/lib/libart.so (art::Runtime::Abort()+160)
I/DEBUG   (  318):     #06 pc 000a5db7  /system/lib/libart.so (art::LogMessage::~LogMessage()+1322)
I/DEBUG   (  318):     #07 pc 000af499  /system/lib/libart.so (art::JniAbort(char const*, char const*)+1076)
I/DEBUG   (  318):     #08 pc 000af9f1  /system/lib/libart.so (art::JniAbortF(char const*, char const*, ...)+60)
I/DEBUG   (  318):     #09 pc 000b05e3  /system/lib/libart.so (art::ScopedCheck::CheckVirtualMethod(_jobject*, _jmethodID*)+402)
I/DEBUG   (  318):     #10 pc 000ba057  /system/lib/libart.so (art::CheckJNI::CallVoidMethodV(_JNIEnv*, _jobject*, _jmethodID*, std::__va_list)+70)
I/DEBUG   (  318):     #11 pc 0006028f  /system/lib/libandroid_runtime.so
I/DEBUG   (  318):     #12 pc 000721e5  /data/dalvik-cache/arm/system@framework@boot.oat
```

**Comment by yujincheng08** [comment]
Try to delete `/data/dalvik-cache/arm/system@framework@boot.oat`

**Comment by Unimog411** [comment]
@yujincheng08 
Thanks for the hint. I tried to delete `/data/dalvik-cache/arm/system@framework@boot.oat` and then uninstalled v23.0 and tried to install Canary (24102) again without success. I tried again with V23 and this couldn't be installed either now. I had to factory reset the HTC One M7 and restart from scratch. Even now, canary 24102 UI crashes when opened.
Sorry that I could not help you with debugging this issue and thanks for the great work you guys are doing...
